awscli/formatter.py

```diff
@@ -71,6 +71,7 @@ def __call__(self, command_name, response, stream=None):
             # so that if anything wraps stdout we'll pick up those changes
             # (specifically colorama on windows wraps stdout).
             stream = self._get_default_stream()
+        compat.set_preferred_output_encoding(stream)
```

Comment:
Is this line and the same one below necessary if `self._get_default_stream()` calls `compat.get_stdout_text_writer()` (and `compat._get_text_writer()`) and returns an already configured stream writer?

Reply:
I tried that at first, however for the case where the response is paginated, the stream we pass into the formatter doesn't come from these code paths, rather within `OutputStreamFactory` we get `stdin` from the pager process here:
https://github.com/aws/aws-cli/blob/882690d812c4ede0fc055fc5949ca85faa1c8fa8/awscli/utils.py#L412-L419

So setting the encdoing from the formatter covers the pagination, but setting it `get_stdout_text_writer`/etc. is used for cases we're not going through the formatters (i.e. error handling).